File: backend/Tour_Travel_api/Tour_Travel_api/Models/SupportTicket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TravelApp.Models
{
    public class SupportTicket
    {
        public int Id { get; set; }
        
        [Required]
        public int UserId { get; set; }
        
        [Required]
        public string Subject { get; set; }
        
        public bool IsClosed { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [JsonIgnore]
        public User User { get; set; }
        
        [JsonIgnore]
        public ICollection<ChatMessage> Messages { get; set; }
    }
}

File: front-end/tour_and_travel/lib/models/support_ticket.dart
import 'package:tour_and_travel/models/chat_message.dart';

class SupportTicket {
  final int id;
  final String subject;
  final bool isClosed;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? userName;
  final String? userEmail;
  final bool hasUnread;
  final List<ChatMessage> messages;

  SupportTicket({
    required this.id,
    required this.subject,
    required this.isClosed,
    required this.createdAt,
    required this.updatedAt,
    this.userName,
    this.userEmail,
    this.hasUnread = false,
    this.messages = const [],
  });

  factory SupportTicket.fromJson(Map<String, dynamic> json) {
    List<ChatMessage> msgList = [];
    if (json['messages'] != null) {
      msgList = (json['messages'] as List)
          .map((m) => ChatMessage.fromJson(m))
          .toList();
    }

    return SupportTicket(
      id: json['id'] ?? 0,
      subject: json['subject'] ?? '',
      isClosed: json['isClosed'] ?? false,
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null 
          ? DateTime.parse(json['updatedAt']) 
          : DateTime.now(),
      userName: json['userName'],
      userEmail: json['userEmail'],
      hasUnread: json['hasUnread'] ?? false,
      messages: msgList,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'subject': subject,
      'isClosed': isClosed,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'userName': userName,
      'userEmail': userEmail,
      'hasUnread': hasUnread,
      'messages': messages.map((m) => m.toJson()).toList(),
    };
  }
}
